Sources/UI/ScreenViewController.swift

```diff
@@ -425,20 +425,63 @@ open class ScreenViewController: UICollectionViewController, UICollectionViewDat
         }
         
         var userInfo: [String: Any] = [
-            "experience": experience.attributes,
-            "screen": screen.attributes,
-            "row": row.attributes,
-            "block": block.attributes
+            ScreenViewController.experienceUserInfoKey: experience,
+            ScreenViewController.screenUserInfoKey: screen,
+            ScreenViewController.blockUserInfoKey: block
         ]
         
-        if let campaignID = self.campaignID {
-            userInfo["campaignID"] = campaignID
+        if let campaignID = campaignID {
+            userInfo[ScreenViewController.campaignIDUserInfoKey] = campaignID
         }
         
         NotificationCenter.default.post(
-            name: .RVBlockTapped,
+            name: ScreenViewController.blockTappedNotification,
             object: self,
             userInfo: userInfo
         )
     }
 }
+
+extension ScreenViewController {
+    /// The `ScreenViewController` sends this notification when it is presented.
+    public static let screenPresentedNotification = Notification.Name("io.rover.screenPresentedNotification")
+    
+    /// The `ScreenViewController` sends this notification when it is dismissed.
+    public static let screenDismissedNotification = Notification.Name("io.rover.screenDismissedNotification")
+    
+    /// The `ScreenViewController` sends this notification when a user finishes viewing a screen. The user starts
+    /// viewing a screen when the view controller is presented and finishes when it is dismissed. The duration the
+    /// user viewed the screen is included in the `durationUserInfoKey`.
+    ///
+    /// If the user quickly dismisses the view controller and presents it again (or backgrounds the app and restores it)
+    /// the view controller considers this part of the same "viewing session". The notification is not sent until the
+    /// user dismisses the view controller and a specified time passes (default is 15 seconds).
+    ///
+    /// This notification is useful for tracking the amount of time users spend viewing a screen. However if you want to
+    /// be notified immediately when a user views a screen you should use the `screenPresentedNotification`.
+    public static let screenViewedNotification = Notification.Name("io.rover.screenViewedNotification")
+    
+    /// The `ScreenViewController` sends this when a `UIView` representing a specific block somewhere within the view
+    /// controller's hierarchy was tapped by the user.
+    public static let blockTappedNotification = Notification.Name("io.rover.blockTappedNotification")
+}
```

Comment:
Is this that new convention for defining notification names that you were talking about?

Reply:
Yup.